lib/parse.js

```diff
@@ -566,45 +579,54 @@ define(['esprima', './lang'], function (esprima, lang) {
      * Finds all dependencies specified in dependency arrays and inside
      * simplified commonjs wrappers.
      * @param {String} fileName
-     * @param {String} fileContents
+     * @param {String} fileContents (or astRoot)
+     * @param {Object} options
      *
      * @returns {Array} an array of dependency strings. The dependencies
-     * have not been normalized, they may be relative IDs.
+     * have not been normalized, they may be relative IDs. An additional
+     * array of Formal parameter names is available as the property "params"
+     * on the returned array.
      */
     parse.findDependencies = function (fileName, fileContents, options) {
         var dependencies = [],
-            astRoot = esprima.parse(fileContents);
+            params = [],
+            astRoot = typeof fileContents === 'string' ?
+                      esprima.parse(fileContents) : fileContents;
 
         parse.recurse(astRoot, function (callName, config, name, deps) {
             if (deps) {
                 dependencies = dependencies.concat(deps);
+                if (deps.params) {
+                    params = params.concat(deps.params);
+                }
             }
         }, options);
 
+        dependencies.params = params;
         return dependencies;
     };
 
     /**
      * Finds only CJS dependencies, ones that are the form
      * require('stringLiteral')
+     * @param {String} fileName
+     * @param {String} fileContents (or astRoot)
+     * @param {Object} options
```

Comment:
This signature does not appear to take an `options` parameter.

Reply:
Ah, my mistake, sorry. `findDependencies` has the options, `findCjsDependencies` not. I'm removing it.